backend/src/libs/packages/open-ai/open-ai.package.ts

```diff
@@ -0,0 +1,76 @@
+import { type HTTPService } from '../http/http.js';
+import {
+  BASE_OPEN_AI_URL,
+  DEFAULT_IMAGE_GENERATE_REQUEST,
+  EMPTY_ARRAY_LENGTH,
+} from './libs/constants/constants.js';
+import {
+  type OpenAiImageGenerateRequestDto,
+  type OpenAiImageGenerateResponseDto,
+  type OpenAiMessageGenerateRequestDto,
+  type OpenAiMessageGenerateResponseDto,
+} from './libs/types/types.js';
+
+type OpenAiDependencies = {
+  httpService: HTTPService;
+  apiKey: string;
+  model?: string;
+};
+
+class OpenAi {
+  private apiKey: string;
+  private httpService: HTTPService;
+  private model: string;
+
+  public constructor({
+    httpService,
+    apiKey,
+    model = 'gpt-3.5-turbo-0301',
```

Comment:
will this model fit for both images and text?

Reply:
we only use the model for message answers - image generation doesn't require model specification

Reply:
got this